apps/server/src/modules/learnroom/uc/course-rooms.uc.ts

```diff
@@ -23,6 +24,11 @@ export class CourseRoomsUc {
 		const user = await this.userService.getUserEntityWithRoles(userId);
 		// TODO no authorisation check here?
 		const course = await this.courseService.findOneForUser(roomId, userId);
+
+		if (course.teachers.length === 0) {
+			throw new LockedCourseLoggableException(course.name, course.id);
+		}
+
```

Comment:
It looks like we have the "business logic" that decides whether a course is locked or not in `Course.getMetaData()`. How about writing `if (course.getMetadata().isLocked)`?